README.md
# markdowndoc-markdown-parser

[![npm](https://img.shields.io/npm/v/markdowndoc-markdown-parser.svg?style=flat-square)](https://www.npmjs.com/package/markdowndoc-markdown-parser)
[![GitHub release](https://img.shields.io/github/release/MarkdownDoc/markdowndoc-markdown-parser.svg?style=flat-square)](https://github.com/MarkdownDoc/markdowndoc-markdown-parser/releases)
[![devDependency Status](https://david-dm.org/MarkdownDoc/markdowndoc-markdown-parser/dev-status.svg?style=flat-square)](https://david-dm.org/MarkdownDoc/markdowndoc-markdown-parser#info=devDependencies)
[![Dependency Status](https://david-dm.org/MarkdownDoc/markdowndoc-markdown-parser.svg?style=flat-square)](https://david-dm.org/MarkdownDoc/markdowndoc-markdown-parser#info=dependencies&view=table)
[![Software License](https://img.shields.io/badge/license-MIT-brightgreen.svg?style=flat-square)](LICENSE)

## Master
[![Build Status](https://img.shields.io/travis/MarkdownDoc/markdowndoc-markdown-parser/master.svg?style=flat-square)](https://travis-ci.org/MarkdownDoc/markdowndoc-markdown-parser)

## Develop
[![Build Status](https://img.shields.io/travis/MarkdownDoc/markdowndoc-markdown-parser/master.svg?style=flat-square)](https://travis-ci.org/MarkdownDoc/markdowndoc-markdown-parser)

### Bits and Pieces:
* [npm:](https://npmjs.org/) Well... it's a Node.js package manager
* [Grunt:](http://gruntjs.com/) Automates common tasks: test, build, clean
* [Babel:](https://github.com/babel/babel/) Use next generation JavaScript, today.

## Getting Started:

1. You need to have npm installed (it comes with node)
2. Clone this repository: `git clone git@github.com:MarkdownDoc/markdowndoc-markdown-parser.git`
3. Install dependencies: `npm install`

## Install

~~~
npm install
bower install
grunt
~~~

## Change log

Please see [CHANGELOG](CHANGELOG.md) for more information what has changed recently.

## Testing

~~~
grunt test
~~~

## Contributing

Please see [CONTRIBUTING](CONTRIBUTING.md) for details.

## Credits

- [Daniel Bannert](https://github.com/prisis),
- [gwagroves](https://github.com/gwagroves)
- [All Contributors](../../contributors)

## License

The MIT License (MIT). Please see [License File](LICENSE.md) for more information.


CHANGELOG.md
# Changelog

All Notable changes to `markdowndoc-markdown-parser` will be documented in this file

## NEXT - YYYY-MM-DD

### Added
- Nothing

### Deprecated
- Nothing

### Fixed
- Nothing

### Removed
- Nothing

### Security
- Nothing


tests/specs/parser.js
'use strict';

var _interopRequireDefault = require('babel-runtime/helpers/interop-require-default')['default'];

exports.__esModule = true;

var _readFile = require('read-file');

var _readFile2 = _interopRequireDefault(_readFile);

var _highlightJs = require('highlight.js');

var _highlightJs2 = _interopRequireDefault(_highlightJs);

var _markdownIt = require('markdown-it');

var _markdownIt2 = _interopRequireDefault(_markdownIt);

var _markdownItCheckbox = require('markdown-it-checkbox');

var _markdownItCheckbox2 = _interopRequireDefault(_markdownItCheckbox);

var _markdownItLinkscheme = require('markdown-it-linkscheme');

var _markdownItLinkscheme2 = _interopRequireDefault(_markdownItLinkscheme);

var _markdownItResponsive = require('markdown-it-responsive');

var _markdownItResponsive2 = _interopRequireDefault(_markdownItResponsive);

exports['default'] = function (options, filepath) {
  function responsiveOption() {
    return {
      responsive: {
        'srcset': {
          'header-*': [{
            width: 480,
            rename: {
              suffix: '-small'
            }
          }, {
            width: 768,
            rename: {
              suffix: '-medium'
            }
          }]
        },
        'sizes': {
          'header-*': '(min-width: 48em) 33.3vw, 100vw'
        }
      }
    };
  }

  function markdownHighlight(str, lang) {
    if (options.highlight) {
      return '';
    }

    if (lang && _highlightJs2['default'].getLanguage(lang)) {
      try {
        return _highlightJs2['default'].highlight(lang, str).value;
      } catch (__) {
        return ''; // use external default escaping
      }
    }

    try {
      return _highlightJs2['default'].highlightAuto(str).value;
    } catch (__) {
      return ''; // use external default escaping
    }
  }

  function markdown() {
    var md = new _markdownIt2['default']({
      highlight: function highlight(str, lang) {
        return markdownHighlight(options, str, lang);
      }
    });

    if (options.linkscheme) {
      md.use(_markdownItLinkscheme2['default']);
    }

    if (options.responsiveimage) {
      md.use(_markdownItResponsive2['default'], responsiveOption());
    }

    if (options.checkbox) {
      md.use(_markdownItCheckbox2['default']);
    }

    return md;
  }

  function renderHtml() {
    var md = markdown(options);
    var file = _readFile2['default'].sync(filepath, { encoding: 'utf8' });

    return md.render(file, { encoding: 'utf8' });
  }

  return renderHtml(options, filepath);
};

module.exports = exports['default'];
//# sourceMappingURL=parser.js.map


tests/src/parser.test.js
'use strict';

var expect = require('chai').expect;
var assert = require('chai').assert;

var parser = require('../specs/parser');

describe('#parser', function () {
  it('If options and folder path is set, get a html string.', function () {
    var options = {
      'highlight': true,
      'linkscheme': true,
      'responsiveimage': true,
      'checkbox': true
    },
    test = parser(options, './tests/fixture/one.md');

    expect(typeof test === 'string').to.equal(true);
    expect(test).to.equal('<h1>one</h1>\n<h1>one2</h1>\n');
  });
});


src/index.js
import read from 'read-file';
import hljs from 'highlight.js';
import MarkdownIt from 'markdown-it';
import checkbox from 'markdown-it-checkbox';
import linkscheme from 'markdown-it-linkscheme';
import responsive from 'markdown-it-responsive';

export default function(options, filepath) {
  function responsiveOption() {
    return {
      responsive: {
        'srcset': {
          'header-*': [{
            width: 480,
            rename: {
              suffix: '-small',
            },
          }, {
            width: 768,
            rename: {
              suffix: '-medium',
            },
          }],
        },
        'sizes': {
          'header-*': '(min-width: 48em) 33.3vw, 100vw',
        },
      },
    };
  }

  function markdownHighlight(str, lang) {
    if (options.highlight) {
      return '';
    }

    if (lang && hljs.getLanguage(lang)) {
      try {
        return hljs.highlight(lang, str).value;
      } catch (__) {
        return ''; // use external default escaping
      }
    }

    try {
      return hljs.highlightAuto(str).value;
    } catch (__) {
      return ''; // use external default escaping
    }
  }

  function markdown() {
    const md = new MarkdownIt({
      highlight: function(str, lang) {
        return markdownHighlight(options, str, lang);
      },
    });

    if (options.linkscheme) {
      md.use(linkscheme);
    }

    if (options.responsiveimage) {
      md.use(responsive, responsiveOption());
    }

    if (options.checkbox) {
      md.use(checkbox);
    }

    return md;
  }

  function renderHtml() {
    const md = markdown(options);
    const file = read.sync(filepath, { encoding: 'utf8' });

    return md.render(file, { encoding: 'utf8' });
  }

  return renderHtml(options, filepath);
}
